ts/Series/DataModifyComposition.ts

```diff
@@ -0,0 +1,805 @@
+            redraw?: boolean
+        ): void;
+    }
+
+    /* *
+     *
+     *  Constants
+     *
+     * */
+
+    const composedClasses: Array<Function> = [];
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    /* eslint-disable valid-jsdoc */
+
+    /**
+     * Extends the series, axis and point classes with
+     * compare and cumulative support.
+     *
+     * @private
+     *
+     * @param SeriesClass
+     * Series class to use.
+     *
+     * @param AxisClass
+     * Axis class to extend.
+     *
+     * @param PointClass
+     * Point class to use.
+     */
+    export function compose<T extends typeof Series>(
+        SeriesClass: T,
+        AxisClass: typeof Axis,
+        PointClass: typeof Point
+    ): (typeof SeriesComposition&T) {
+        if (composedClasses.indexOf(SeriesClass) === -1) {
+            composedClasses.push(SeriesClass);
+
+            const seriesProto = SeriesClass.prototype as SeriesComposition;
+
+            seriesProto.setCompare = seriesSetCompare;
+            seriesProto.setCumulative = seriesSetCumulative;
+
+            addEvent(SeriesClass, 'afterInit', afterInit);
+            addEvent(SeriesClass, 'afterGetExtremes', afterGetExtremes);
+            addEvent(SeriesClass, 'afterProcessData', afterProcessData);
+        }
+
+        if (composedClasses.indexOf(AxisClass) === -1) {
+            composedClasses.push(AxisClass);
+
+            const axisProto = AxisClass.prototype as AxisComposition;
+
+            axisProto.setCompare = axisSetCompare;
+            axisProto.setModifier = setModifier;
+            axisProto.setCumulative = axisSetCumulative;
+        }
+
+        if (composedClasses.indexOf(PointClass) === -1) {
+            composedClasses.push(PointClass);
+
+            const pointProto = PointClass.prototype as PointComposition;
+
+            pointProto.tooltipFormatter = tooltipFormatter;
+        }
+
+        return SeriesClass as (typeof SeriesComposition&T);
+    }
+
+    /* ********************************************************************** *
+     *  Start shared compare and cumulative logic                             *
+     * ********************************************************************** */
+
+    /**
+     * Shared code for the axis.setCompare() and the axis.setCumulative()
+     * methods. Inits the 'compare' or the 'cumulative' mode.
+     * @private
+     */
+    function setModifier(
+        this: Axis,
+        mode: 'compare'|'cumulative',
+        modeState?: boolean|null|'percent'|'value',
+        redraw?: boolean
+    ): void {
+        if (!this.isXAxis) {
+            this.series.forEach(function (series): void {
+                if (
+                    mode === 'compare' &&
+                    (
+                        modeState === 'percent' ||
+                        modeState === 'value' ||
+                        typeof modeState === 'undefined' ||
+                        modeState === null
```

Comment:
I might be missing something, but do we need the second part of this `if`, which checks all possible types of `modeState`?  The same below, where we would check types for `cumulative` mdoe.